### DISCUSSION TITLE
Does gortsplib support IPv6 ?

### DISCUSSION THREAD

**Discussion by tifler** [bluenviron/gortsplib#183] [Questions]
Hi,

I'm considering this project to build my service which needs IPv6 support but I'm not sure whether the project supports IPv6 or not.
Anyone who knows it, please let me know.

Thanks in advance.

**Comment by aler9** [comment]
Hello, this project supports IPv6 and creates two listeners by default, a IPv4 one and a IPv6 one. If you need to fine-tune the listen address you can explicitly set it inside server options:

```go
s := &gortsplib.Server{
	Handler:           &serverHandler{},
	RTSPAddress:       "[0087:680f:ec4a:6330:434d:0c33:cd5f:49f3]:8554",
	UDPRTPAddress:     "[0087:680f:ec4a:6330:434d:0c33:cd5f:49f3]:8000",
	UDPRTCPAddress:    "[0087:680f:ec4a:6330:434d:0c33:cd5f:49f3]:8001",
}
```

or client options:

```go
c := gortsplib.Client{}
...
u, err := url.Parse("rtsp://[0087:680f:ec4a:6330:434d:0c33:cd5f:49f3]:8554/mystream")
...
err = c.Start(u.Scheme, u.Host)
```
